Assignment2/tests/game.test.ts

```diff
@@ -0,0 +1,90 @@
+import { expect, test } from "vitest";
+import { Game } from "~/scripts/game";
+import { WordList } from "~/scripts/wordList";
+import { updateValidWords } from "~/scripts/wordListUtils";
+
+//These first two were here before, accidentally deleted at some point ig
+test("game", () => {
+    // create game and check if it's created
+    const game = new Game(6);
+    expect(game.secretWord.length).toBe(5);
+    expect(game.maxAttempts).toBe(6);
+    expect(updateValidWords(game)).toContain(game.secretWord.toLowerCase());
+});
+
+test("guess-word", () => {
+    const game = new Game(6);
+    game.guess.addLetter("Z");
+    game.guess.addLetter("Z");
+    game.guess.addLetter("Z");
+    game.guess.addLetter("Z");
+    game.guess.addLetter("Z");
+    game.guess.addLetter("Z");
+    expect(game.guess.letters.length).toBe(5);
+    expect(game.guess.word).toBe("ZZZZZ");
+});
+
+
+test("wordList init", () => {
+  const game = new Game(6);
+  expect(updateValidWords(game).length).equal(WordList.length);
+});
+
+test("none right", () => {
+  const game = new Game(6);
+  game.secretWord = "HELLO";
+  game.guess.addLetter("A");
+  game.guess.addLetter("P");
+  game.guess.addLetter("A");
+  game.guess.addLetter("R");
+  game.guess.addLetter("T");
+  game.submitGuess();
+
+  expect(updateValidWords(game)).not.toContain("hands");
+  expect(updateValidWords(game)).not.toContain("snare");
+  expect(updateValidWords(game)).not.toContain("traps");
+  expect(updateValidWords(game)).not.toContain("apart");
+  expect(updateValidWords(game)).not.toContain("aptly");
+
+  expect(updateValidWords(game)).toContain("hello");
+  expect(updateValidWords(game)).toContain("holds");
+  expect(updateValidWords(game)).toContain("smell");
+  expect(updateValidWords(game)).toContain("messy");
+  expect(updateValidWords(game)).toContain("elves");
```

Comment:
Extensive! While manual tests are sufficient, consider ways to test this logic in more general cases